From c59ed7d236586867d3deb51b7b612d946e758365 Mon Sep 17 00:00:00 2001
From: Muhammad Saheed <saheed@FreeBSD.org>
Date: Sun, 21 Sep 2025 02:14:46 +0530
Subject: [PATCH] sysutils/wutil: Add new port

wutil provides a CLI and TUI for WiFi management. It interacts with
wpa_supplicant over its control interface.
---
 sysutils/wutil/Makefile  | 31 +++++++++++++++++++++++++++++++
 sysutils/wutil/distinfo  |  3 +++
 sysutils/wutil/pkg-descr |  2 ++
 3 files changed, 36 insertions(+)
 create mode 100644 sysutils/wutil/Makefile
 create mode 100644 sysutils/wutil/distinfo
 create mode 100644 sysutils/wutil/pkg-descr

diff --git a/sysutils/wutil/Makefile b/sysutils/wutil/Makefile
new file mode 100644
index 0000000000..75335a872a
--- /dev/null
+++ b/sysutils/wutil/Makefile
@@ -0,0 +1,31 @@
+PORTNAME=	wutil
+DISTVERSIONPREFIX=	v
+DISTVERSION=	0.3.0
+CATEGORIES=	sysutils
+
+MAINTAINER=	saheed@FreeBSD.org
+COMMENT=	CLI and TUI for WiFi Management
+WWW=		https://github.com/MainKt/wutil
+
+LICENSE=	BSD2CLAUSE
+LICENSE_FILE=	${WRKSRC}/LICENSE
+
+LIB_DEPENDS+=	libifconfig.so:net/libifconfig
+
+MAKE_ARGS+=	CFLAGS="-I${LOCALBASE}/include" LDFLAGS="-L${LOCALBASE}/lib"
+
+USE_GITHUB=	yes
+GH_ACCOUNT=	MainKt
+
+PLIST_FILES=	bin/wutil \
+							bin/wutui \
+							share/man/man8/wutil.8.gz \
+							share/man/man8/wutui.8.gz
+
+do-install:
+	${INSTALL_PROGRAM} ${WRKSRC}/wutil ${STAGEDIR}${PREFIX}/bin
+	${INSTALL_PROGRAM} ${WRKSRC}/wutui ${STAGEDIR}${PREFIX}/bin
+	${INSTALL_MAN} ${WRKSRC}/wutil.8 ${STAGEDIR}${PREFIX}/share/man/man8
+	${INSTALL_MAN} ${WRKSRC}/wutui.8 ${STAGEDIR}${PREFIX}/share/man/man8
+
+.include <bsd.port.mk>
diff --git a/sysutils/wutil/distinfo b/sysutils/wutil/distinfo
new file mode 100644
index 0000000000..b9b56f96b0
--- /dev/null
+++ b/sysutils/wutil/distinfo
@@ -0,0 +1,3 @@
+TIMESTAMP = 1758400904
+SHA256 (MainKt-wutil-v0.3.0_GH0.tar.gz) = c3883d0f264e868c90da0cb3323aa36c27594058eb2608c3d1a929fca3696be7
+SIZE (MainKt-wutil-v0.3.0_GH0.tar.gz) = 40403
diff --git a/sysutils/wutil/pkg-descr b/sysutils/wutil/pkg-descr
new file mode 100644
index 0000000000..42cdebf0d4
--- /dev/null
+++ b/sysutils/wutil/pkg-descr
@@ -0,0 +1,2 @@
+wutil is a WiFi management utility for FreeBSD.
+It provides a CLI, wutil, and a TUI, wutui.
-- 
2.49.0

